orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java

```diff
@@ -0,0 +1,559 @@
+import org.hibernate.ogm.model.spi.Tuple;
+import org.hibernate.ogm.persister.impl.OgmCollectionPersister;
+import org.hibernate.ogm.persister.impl.OgmEntityPersister;
+import org.hibernate.ogm.type.spi.GridType;
+import org.hibernate.persister.collection.CollectionPersister;
+import org.hibernate.persister.entity.EntityPersister;
+import org.hibernate.type.Type;
+
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import org.hibernate.datastore.ogm.orientdb.constant.OrientDBConstant;
+import org.hibernate.datastore.ogm.orientdb.dialect.impl.OrientDBAssociationSnapshot;
+import org.hibernate.datastore.ogm.orientdb.dialect.impl.OrientDBTupleAssociationSnapshot;
+import org.hibernate.datastore.ogm.orientdb.dialect.impl.ResultSetTupleIterator;
+import org.hibernate.datastore.ogm.orientdb.impl.OrientDBSchemaDefiner;
+import org.hibernate.datastore.ogm.orientdb.type.spi.ORidBagGridType;
+import org.hibernate.datastore.ogm.orientdb.utils.AssociationUtil;
+import org.hibernate.datastore.ogm.orientdb.utils.EntityKeyUtil;
+import org.hibernate.ogm.dialect.query.spi.QueryableGridDialect;
+import org.hibernate.ogm.model.key.spi.RowKey;
+import org.hibernate.service.spi.ServiceRegistryAwareService;
+import org.hibernate.service.spi.ServiceRegistryImplementor;
+
+/**
+ * @author Sergey Chernolyas (sergey.chernolyas@gmail.com)
+ */
+public class OrientDBDialect extends BaseGridDialect implements MultigetGridDialect, QueryableGridDialect<String>,
+		ServiceRegistryAwareService, SessionFactoryLifecycleAwareDialect, IdentityColumnAwareGridDialect {
+
+	private static final Log log = LoggerFactory.getLogger();
+	private static final Association ASSOCIATION_NOT_FOUND = null;
+
+	private OrientDBDatastoreProvider provider;
+	private ServiceRegistryImplementor serviceRegistry;
+	private Map<AssociationKeyMetadata, OrientDBAssociationQueries> associationQueries;
+	private Map<EntityKeyMetadata, OrientDBEntityQueries> entityQueries;
+
+	public OrientDBDialect(OrientDBDatastoreProvider provider) {
+		this.provider = provider;
+	}
+
+	@Override
+	public Tuple getTuple(EntityKey key, TupleContext tupleContext) {
+		log.info( "getTuple:EntityKey:" + key + "; tupleContext" + tupleContext + " tupleContext.getClass():" + tupleContext.getClass() );
+
+		try {
+			if ( !entityQueries.containsKey( key.getMetadata() ) ) {
+				// throw new EntityNotFoundException("Key : "+key.toString());
+				return null;
+			}
+			Map<String, Object> dbValuesMap = entityQueries.get( key.getMetadata() ).findEntity( provider.getConnection(), key );
+			if ( dbValuesMap == null || ( dbValuesMap != null && dbValuesMap.isEmpty() ) ) {
+				return null;
+			}
+			return new Tuple(
+					new OrientDBTupleSnapshot( dbValuesMap, tupleContext.getAllAssociatedEntityKeyMetadata(), tupleContext.getAllRoles(), key.getMetadata() ) );
+		}
+		catch (SQLException e) {
+			log.error( "Can not find entity", e );
```

Comment:
SQL exception.  About "entity not found" ... You are right. The code maust throw "entity not found"  exception. I will fix the incorrection